mobile/android/android-device-hash-service.ts

```diff
@@ -52,22 +52,32 @@ export class AndroidDeviceHashService implements Mobile.IAndroidDeviceHashServic
 		return false;
 	}
 
-	public async generateHashesFromLocalToDevicePaths(localToDevicePaths: Mobile.ILocalToDevicePathData[], shasums: IStringDictionary): Promise<string[]> {
+	public async generateHashesFromLocalToDevicePaths(localToDevicePaths: Mobile.ILocalToDevicePathData[], initialShasums: IStringDictionary = {}): Promise<IStringDictionary> {
 		const devicePaths: string[] = [];
 		const action = async (localToDevicePathData: Mobile.ILocalToDevicePathData) => {
 			const localPath = localToDevicePathData.getLocalPath();
 			if (this.$fs.getFsStats(localPath).isFile()) {
 				// TODO: Use relative to project path for key
 				// This will speed up livesync on the same device for the same project on different PCs.
-				shasums[localPath] = await this.$fs.getFileShasum(localPath);
+				initialShasums[localPath] = await this.$fs.getFileShasum(localPath);
 			}
 
 			devicePaths.push(`"${localToDevicePathData.getDevicePath()}"`);
 		};
 
 		await executeActionByChunks<Mobile.ILocalToDevicePathData>(localToDevicePaths, DEFAULT_CHUNK_SIZE, action);
 
-		return devicePaths;
+		return initialShasums;
+	}
+
+	public getDevicePaths(localToDevicePaths: Mobile.ILocalToDevicePathData[]): string[] {
```

Comment:
We can merge `getDevicePaths` and `generateHashesFromLocalToDevicePaths` (by returning another interface) in order to avoid iterating `localToDevicePaths` twice.

Reply:
This will make the name of the method generate HashesAndDevicePaths.... which raises a red flag for me. It can be implemented, but my personal preference is to remain as separate methods.

Also getDevicePaths is used only in one place, and generateHashesFromLocalToDevicePaths in 3, which means more often we don't really need them.